From 559e6c00a1201d56204ee774b6f63ea85331bdb2 Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Mon, 5 Feb 2024 12:12:17 +0000
Subject: [PATCH 0882/1085] drivers: media: cfe: Increase default size of
 embedded buffer

Increase the size of the default embedded buffer to 16k. This is done to
match what is advertised by the IMX219 driver and workaround a problem
where the embedded stream is not actually used. Without full streams API
support, the media pipeline validation will fail in these circumstances.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/platform/raspberrypi/rp1_cfe/cfe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/media/platform/raspberrypi/rp1_cfe/cfe.c
+++ b/drivers/media/platform/raspberrypi/rp1_cfe/cfe.c
@@ -93,7 +93,7 @@ MODULE_PARM_DESC(verbose_debug, "verbose
 #define MIN_WIDTH 16
 #define MIN_HEIGHT 16
 /* Default size of the embedded buffer */
-#define DEFAULT_EMBEDDED_SIZE 8192
+#define DEFAULT_EMBEDDED_SIZE 16384
 
 const struct v4l2_mbus_framefmt cfe_default_format = {
 	.width = 640,
@@ -107,7 +107,7 @@ const struct v4l2_mbus_framefmt cfe_defa
 };
 
 const struct v4l2_mbus_framefmt cfe_default_meta_format = {
-	.width = 8192,
+	.width = DEFAULT_EMBEDDED_SIZE,
 	.height = 1,
 	.code = MEDIA_BUS_FMT_SENSOR_DATA,
 	.field = V4L2_FIELD_NONE,
